pyomo/contrib/gdpopt/branch_and_bound.py

```diff
@@ -37,442 +45,470 @@
-            from_list=subproblem.GDPopt_utils.variable_list,
-            to_list=model.GDPopt_utils.variable_list,
-            config=config, ignore_integrality=True
-        )
-        return float('-inf'), float('inf')
-
-    term_cond = result.solver.termination_condition
-    if term_cond == tc.optimal:
-        assert result.solver.status is SolverStatus.ok
-        lb = result.problem.lower_bound if not obj_sense_correction else \
-             -result.problem.upper_bound
-        ub = result.problem.upper_bound if not obj_sense_correction else \
-             -result.problem.lower_bound
-        copy_var_list_values(
-            from_list=subproblem.GDPopt_utils.variable_list,
-            to_list=model.GDPopt_utils.variable_list,
-            config=config,
-        )
-        return float('-inf'), ub
-    elif term_cond == tc.locallyOptimal or term_cond == tc.feasible:
-        assert result.solver.status is SolverStatus.ok
-        lb = result.problem.lower_bound if not obj_sense_correction else \
-             -result.problem.upper_bound
-        ub = result.problem.upper_bound if not obj_sense_correction else \
-             -result.problem.lower_bound
-        # TODO handle LB absent
-        copy_var_list_values(
-            from_list=subproblem.GDPopt_utils.variable_list,
-            to_list=model.GDPopt_utils.variable_list,
-            config=config,
-        )
-        return float('-inf'), ub
-    elif term_cond == tc.unbounded:
-        copy_var_list_values(
-            from_list=subproblem.GDPopt_utils.variable_list,
-            to_list=model.GDPopt_utils.variable_list,
-            config=config, ignore_integrality=True
-        )
-        return float('-inf'), float('-inf')
-    elif term_cond == tc.infeasible:
-        copy_var_list_values(
-            from_list=subproblem.GDPopt_utils.variable_list,
-            to_list=model.GDPopt_utils.variable_list,
-            config=config, ignore_integrality=True
-        )
-        return float('-inf'), float('inf')
-    else:
-        config.logger.warning("Unknown termination condition of %s. "
-                              "Treating as infeasible." % term_cond)
-        copy_var_list_values(
-            from_list=subproblem.GDPopt_utils.variable_list,
-            to_list=model.GDPopt_utils.variable_list,
-            config=config, ignore_integrality=True
-        )
-        return float('-inf'), float('inf')
+            config.logger.warning("Unknown termination condition of %s. "
+                                  "Treating as infeasible." % term_cond)
+            copy_var_list_values(
+                from_list=subprob_utils.algebraic_variable_list,
+                to_list=model_utils.algebraic_variable_list,
+                config=config, ignore_integrality=True
+            )
+            return float('inf'), float('inf')
+
+
+    def _solve_local_rnGDP_subproblem(self, model, config):
+        # TODO for now, return (LB, UB) = (-inf, inf) (for minimize)
```

Comment:
This could be addressed now

Reply:
I don't actually understand this TODO... What are you thinking should be changed? In general I think we should save a rewrite of this algorithm for a separate PR, but I'm happy to make some small changes.

Reply:
Ah, okay, I will clarify the comment then. Thanks!